File: Contests/Div4/849/B.py
def solve():
    n = int(input())
    s = input()
    eD = cB = 0
    for i in range(n):
        if (eD == 1 and cB == 1):
            print("YES")
            return
        if s[i] == "L":
            eD -= 1
        elif s[i] == "R":
            eD += 1
        elif s[i] == "D":
            cB -= 1
        else:
            cB += 1
    if (eD == 1 and cB == 1):
        print("YES")
    else:
        print("NO")
        
t = int(input())
while (t>0):
    solve()
    t -= 1

File: Contests/Div4/849/B.cpp
#include <bits/stdc++.h>

using namespace std;

void solve(){
    int n;
    cin >> n;
    string s;
    cin >> s;
    int cB = 0, dE = 0;
    for (int i = 0 ; i < n ; i++){
        if (cB == 1 && dE == 1){
            cout << "YES" << '\n';
            return;
        }
        if (s[i] == 'U') cB++;
        else if (s[i] == 'D') cB--;
        else if(s[i] == 'L') dE --;
        else dE++;

    }
    if (cB == 1 && dE == 1){
        cout << "YES" << '\n';
     
    }else cout << "NO" << '\n';
    
}

int main(void){
    ios_base::sync_with_stdio(false); cin.tie(0);
    int t; cin >> t; while(t--){solve();}
}